### ISSUE TITLE
Audit consumer proguard rules

### ISSUE THREAD

**Issue by samtstern** [firebase/FirebaseUI-Android#878]
cc @ashwinraghav who has been helping a developer debug a proguard issue

In https://github.com/firebase/FirebaseUI-Android/pull/573 the following lines were introduced to the consumer proguard rules:

```
// REMOVED
-dontwarn retrofit.**		
-dontwarn retrofit2.**		

// ADDED
-dontnote retrofit2.Platform
-dontnote retrofit2.Platform$IOS$MainThreadExecutor
-dontwarn retrofit2.Platform$Java8
```

Ashwin changed them to this in a branch that seems to solve the developer's issue:

```
-dontwarn okio.** #safe
-dontwarn retrofit2.Call #safe
#-dontnote retrofit2.Platform #unsafe
#-dontnote retrofit2.Platform$IOS$MainThreadExecutor
-dontwarn retrofit2.Platform$Java8 #safe
```

Ashwin can you explain what the issue was?

**Comment by SUPERCILEX** [comment]
As a side note, this is Retrofit `v2.3.0`'s suggested config:
```proguard
# Platform calls Class.forName on types which do not exist on Android to determine platform.
-dontnote retrofit2.Platform
# Platform used when running on Java 8 VMs. Will not be used at runtime.
-dontwarn retrofit2.Platform$Java8
# Retain generic type information for use by reflection by converters and adapters.
-keepattributes Signature
# Retain declared checked exceptions for use by a Proxy instance.
-keepattributes Exceptions
```

**Comment by samtstern** [comment]
This has been fixed and released in version `3.2.2`.
